src/common/typeGuards.ts

```diff
@@ -0,0 +1,3 @@
+export function isDefined<T>(t: T | undefined): t is T {
```

Comment:
location of this feels weird, but i like the way it simplifies the code that uses it so it stands for now i think